onionshare_gui/onionshare_gui.py

```diff
@@ -389,6 +400,17 @@ def check_for_requests(self):
         """
         self.update()
 
+        # Have we lost connection to Tor somehow?
+        if self.tor_con.t.isFinished():
```

Comment:
I'm not sure, but think this might be the issue.

`self.tor_con` is an instance of `TorConnectionDialog`, which is a dialog window that's supposed to open, connect to Tor using the passed in `onion` object and display output, and then close. Before it wasn't a class variable, and the reference to it died as soon as the `OnionShareGui` constructor finished executing. I think, in Linux (but maybe not in Mac for some reason), as soon as you reference `self.tor_con` here, PyQt5 is for some reason showing the window again after it's been hidden, only all the widgets in it are still hidden.

What are you trying to do here? I think `self.tor_con.t` isn't the right object you should be checking, because it's just an old thread that was used to connect to Tor, and doesn't need to exist anymore. I think everything you need should be in `self.onion`, like maybe `self.onion.c` for the stem Controller, or `self.onion.tor_proc` for the actual tor subprocess.